### ISSUE TITLE
Airbyte without internet

### ISSUE THREAD

**Issue by yash-ghelani-dsw** [airbytehq/airbyte#14997]
Can airbyte run on a server without internet i.e. source and destination resides locally and transformation git path can be locally provided?

**Comment by harshithmullapudi** [comment]
Hey not sure that is gonna work as we will have to pull the images for source/destination from the internet. But could you explain the usecase in more detail?

**Comment by yash-ghelani-dsw** [comment]
> Hey not sure that is gonna work as we will have to pull the images for source/destination from the internet. But could you explain the usecase in more detail?

Hi I have 3-4 different sources which have different kind of data and I wanted to aggregate this data into a single data source after transformation on it. So The problem is my server does have continious internet access. So for the first time (while configuration), I can arrange the internet access.

**Comment by harshithmullapudi** [comment]
yeah, then it should be fine? Do feel free to try. I will be closing this issue but in case you face any issues do feel free to open it up again

**Comment by yash-ghelani-dsw** [comment]
> yeah, then it should be fine? Do feel free to try. I will be closing this issue but in case you face any issues do feel free to open it up again

but during transformation, It requires git url and docker image name (fishtownanalytics/dbt:1.0.0). Does it requires continious internet access for data transfer or only at time of configuration?

**Comment by harshithmullapudi** [comment]
yeah you will be needing then

**Comment by skt7** [comment]
Hello @harshithmullapudi, I am working along with @yash-ghelani-dsw. 

Let me try to rephrase the question. 

Precisely, what are we trying to achieve is - The server where we are installing Airbyte would not have continuous access to the public internet. Hence we are looking into pointers where we feel currently Airbyte uses the internet for some or other functionality. Primarily we were able to identify the below points (followed by possible solutions we think) -

a) git URL for custom dbt transformation 
Sol: using a local git repo

b) pull the images for source/destination
Sol: Download required images by enabling the internet for a short period of time during the initial set-up

c) fishtownanalytics/dbt:1.0.0 docker image required to run dbt transformation
Sol: Download the "fishtownanalytics/dbt:1.0.0" image by enabling the internet for a short period of time during the initial set-up

Can you please help validate if the above solutions would work? 

Also, if you can let us know if there are any other things we need to take care to achieve what we are targeting i.e. running Airbyte without internet.

**Comment by harshithmullapudi** [comment]
Hey yeah, this looks good to me you have considered all possible things. I would suggest you to run this initially with the internet and then turn it off

**Comment by yash-ghelani-dsw** [comment]
Hi @harshithmullapudi 

To add to what @skt7 has mentioned, we were successfully able to close point b) and c) i.e. once the required docker images are downloaded connector runs seamlessly even when internet is put off. 

We are testing for a) by setting up local git repo, we are "fatal: dumb http transport does not support shallow capabilities" issue. Seems it occurs when "git clone --depth 5 --single-branch  $GIT_REPO git_repo" is run. We even enabled https over local, still no luck. Can you please suggest how can we overcome same.

Attaching the error log below: 

```
2022-08-04 11:56:07  [32mINFO [m i.a.w.p.DockerProcessFactory(create):163 - Preparing command: docker run --rm --init -i -w /data/25/1/transform --log-driver none --name normalization-normalize-25-1-cczii --network host -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -e DEPLOYMENT_MODE=OSS -e WORKER_ENVIRONMENT=DOCKER -e AIRBYTE_ROLE= -e AIRBYTE_VERSION=0.39.41-alpha airbyte/normalization:0.2.12 configure-dbt --integration-type postgres --config destination_config.json --git-repo https://6528-103-199-136-124.ngrok.io/test.git
2022-08-04 11:56:08  [42mnormalization [0m > Running: git clone --depth 5 --single-branch  $GIT_REPO git_repo
2022-08-04 11:56:08  [42mnormalization [0m > Cloning into 'git_repo'...
2022-08-04 11:56:09  [42mnormalization [0m > fatal: dumb http transport does not support shallow capabilities
2022-08-04 11:56:10  [32mINFO [m i.a.w.t.TemporalAttemptExecution(lambda$getWorkerThread$2):158 - Completing future exceptionally...
io.airbyte.workers.exception.WorkerException: Dbt Transformation Failed.
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:60) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:18) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:1589) [?:?]
Caused by: io.airbyte.workers.exception.WorkerException: DBT Transformation Failed.
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:57) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	... 3 more
	Suppressed: io.airbyte.workers.exception.WorkerException: Normalization process wasn't successful
		at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:162) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.general.DbtTransformationRunner.close(DbtTransformationRunner.java:131) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:46) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:18) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at java.lang.Thread.run(Thread.java:1589) [?:?]
2022-08-04 11:56:10  [32mINFO [m i.a.w.t.TemporalAttemptExecution(get):131 - Stopping cancellation check scheduling...
2022-08-04 11:56:10  [32mINFO [m i.a.w.t.TemporalUtils(withBackgroundHeartbeat):291 - Stopping temporal heartbeating...
2022-08-04 11:56:10  [33mWARN [m i.t.i.a.POJOActivityTaskHandler(activityFailureToResult):307 - Activity failure. ActivityId=3f74b806-5811-39d9-9658-db8962664f1c, activityType=Run, attempt=1
java.lang.RuntimeException: io.temporal.serviceclient.CheckedExceptionWrapper: java.util.concurrent.ExecutionException: io.airbyte.workers.exception.WorkerException: Dbt Transformation Failed.
	at io.airbyte.workers.temporal.TemporalUtils.withBackgroundHeartbeat(TemporalUtils.java:289) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.sync.DbtTransformationActivityImpl.run(DbtTransformationActivityImpl.java:77) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:578) ~[?:?]
	at io.temporal.internal.activity.POJOActivityTaskHandler$POJOActivityInboundCallsInterceptor.execute(POJOActivityTaskHandler.java:214) ~[temporal-sdk-1.8.1.jar:?]
	at io.temporal.internal.activity.POJOActivityTaskHandler$POJOActivityImplementation.execute(POJOActivityTaskHandler.java:180) ~[temporal-sdk-1.8.1.jar:?]
	at io.temporal.internal.activity.POJOActivityTaskHandler.handle(POJOActivityTaskHandler.java:120) ~[temporal-sdk-1.8.1.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:204) ~[temporal-sdk-1.8.1.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:164) ~[temporal-sdk-1.8.1.jar:?]
	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:93) ~[temporal-sdk-1.8.1.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) [?:?]
	at java.lang.Thread.run(Thread.java:1589) [?:?]
Caused by: io.temporal.serviceclient.CheckedExceptionWrapper: java.util.concurrent.ExecutionException: io.airbyte.workers.exception.WorkerException: Dbt Transformation Failed.
	at io.temporal.serviceclient.CheckedExceptionWrapper.wrap(CheckedExceptionWrapper.java:56) ~[temporal-serviceclient-1.8.1.jar:?]
	at io.temporal.internal.sync.WorkflowInternal.wrap(WorkflowInternal.java:448) ~[temporal-sdk-1.8.1.jar:?]
	at io.temporal.activity.Activity.wrap(Activity.java:51) ~[temporal-sdk-1.8.1.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:135) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.sync.DbtTransformationActivityImpl.lambda$run$3(DbtTransformationActivityImpl.java:108) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalUtils.withBackgroundHeartbeat(TemporalUtils.java:284) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	... 12 more
Caused by: java.util.concurrent.ExecutionException: io.airbyte.workers.exception.WorkerException: Dbt Transformation Failed.
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[?:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:129) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.sync.DbtTransformationActivityImpl.lambda$run$3(DbtTransformationActivityImpl.java:108) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalUtils.withBackgroundHeartbeat(TemporalUtils.java:284) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	... 12 more
Caused by: io.airbyte.workers.exception.WorkerException: Dbt Transformation Failed.
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:60) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:18) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	... 1 more
Caused by: io.airbyte.workers.exception.WorkerException: DBT Transformation Failed.
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:57) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:18) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
	... 1 more
	Suppressed: io.airbyte.workers.exception.WorkerException: Normalization process wasn't successful
		at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:162) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.general.DbtTransformationRunner.close(DbtTransformationRunner.java:131) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:46) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.general.DbtTransformationWorker.run(DbtTransformationWorker.java:18) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:155) ~[io.airbyte-airbyte-workers-0.39.41-alpha.jar:?]
		at java.lang.Thread.run(Thread.java:1589) [?:?]
2022-08-04 11:55:32  [32mINFO [m i.a.w.t.TemporalAttemptExecution(get):105 - Docker volume job log path: /tmp/workspace/25/0/logs.log
2022-08-04 11:55:32  [32mINFO [m i.a.w.t.TemporalAttemptExecution(get):110 - Executing worker wrapper. Airbyte version: 0.39.41-alpha
2022-08-04 11:55:32  [32mINFO [m i.a.c.i.LineGobbler(voidCall):82 - Checking if airbyte/destination-postgres:0.3.21 exists...
2022-08-04 11:55:32  [32mINFO [m i.a.c.i.LineGobbler(voidCall):82 - airbyte/destination-postgres:0.3.21 was found locally.
2022-08-04 11:55:32  [32mINFO [m i.a.w.p.DockerProcessFactory(create):108 - Creating docker job ID: 25
2022-08-04 11:55:32  [32mINFO [m i.a.w.p.DockerProcessFactory(create):163 - Preparing command: docker run --rm --init -i -w /data/25/0 --log-driver none --name destination-postgres-check-25-0-mqxey --network host -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -e DEPLOYMENT_MODE=OSS -e USE_STREAM_CAPABLE_STATE=false -e WORKER_ENVIRONMENT=DOCKER -e AIRBYTE_ROLE= -e WORKER_CONNECTOR_IMAGE=airbyte/destination-postgres:0.3.21 -e WORKER_JOB_ATTEMPT=0 -e AIRBYTE_VERSION=0.39.41-alpha -e WORKER_JOB_ID=25 airbyte/destination-postgres:0.3.21 check --config source_config.json
2022-08-04 11:55:33  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: Class path contains multiple SLF4J bindings.
2022-08-04 11:55:33  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: Found binding in [jar:file:/airbyte/lib/log4j-slf4j-impl-2.17.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-08-04 11:55:33  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: Found binding in [jar:file:/airbyte/lib/slf4j-reload4j-1.7.36.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-08-04 11:55:33  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: Found binding in [jar:file:/airbyte/lib/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-08-04 11:55:33  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: Found binding in [jar:file:/airbyte/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
2022-08-04 11:55:33  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
2022-08-04 11:55:34  [1;31mERROR [m i.a.c.i.LineGobbler(voidCall):82 - SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
```

**Comment by harshithmullapudi** [comment]
Hey yeah this looks to me like issue on the local git I can try looking at it this EOW

**Comment by yash-ghelani-dsw** [comment]
> Hey yeah this looks to me like issue on the local git I can try looking at it this EOW

Hi @harshithmullapudi , did you get time to look into the local git repo issue?

**Comment by harshithmullapudi** [comment]
Hey yeah looks like --depth is not supported by the HTTP clone option maybe you have make some changes in the dbt image and use that custom image.

**Comment by yash-ghelani-dsw** [comment]
> Hey yeah looks like --depth is not supported by the HTTP clone option maybe you have make some changes in the dbt image and use that custom image.

hi @harshithmullapudi , i have tried with https local server also but it does not work.... Now as per your suggestion I tried to customize the dbt(airbyte/normalization) image, but i failed to get inside the image (after running it as container).
I tried using following command:
"sudo docker run -it airbyte/normalization:0.2.6 /bin/sh"
but it gives following error:
"Unknown command: sh"
So can u help me with following 2 things:
1) can u confirm Airbyte/normalization image is responsible for cloning the git repo...so that customization needs to be done on same
2) Is there any specific way to customize the above image because i believe it is built in such a way that it cannot be run as a container and without running as container, how can i customize it?

**Comment by yash-ghelani-dsw** [comment]
> Hey yeah looks like --depth is not supported by the HTTP clone option maybe you have make some changes in the dbt image and use that custom image.

hi @harshithmullapudi , I was able to find some solution to change the airbyte/normalization image. I have tried to edit following lines in the entrypoint.sh :
""""
if [[ -z "${GIT_BRANCH}" ]]; then
      # Checkout a particular branch from the git repository
      echo "Running: git clone --depth ${GIT_HISTORY_DEPTH} --single-branch  \$GIT_REPO git_repo"
      git clone [[--depth ${GIT_HISTORY_DEPTH} --single-branch]]  "${GIT_REPO}" git_repo
    else
      # No git branch specified, use the default branch of the git repository
      echo "Running: git clone --depth ${GIT_HISTORY_DEPTH} -b ${GIT_BRANCH} --single-branch  \$GIT_REPO git_repo"
      git clone [[--depth ${GIT_HISTORY_DEPTH} -b "${GIT_BRANCH}" --single-branch]]  "${GIT_REPO}" git_repo
    fi
    # Print few history logs to make it easier for users to verify the right code version has been checked out from git
    echo "Last 5 commits in git_repo:"
    (cd git_repo; git log --oneline -${GIT_HISTORY_DEPTH}; cd -)
 """"
I have remove the lines in [[ ]] but it still not working..it shows following error:

"2022-08-20 16:47:43 - Additional Failure Information: message='io.temporal.serviceclient.CheckedExceptionWrapper: java.util.concurrent.ExecutionException: io.airbyte.workers.exception.WorkerException: Normalization Failed.', type='java.lang.RuntimeException', nonRetryable=false ". i have also attached whole error log file.

[logs-159.txt](https://github.com/airbytehq/airbyte/files/9387737/logs-159.txt)

can u please help me what are the changes required to remove the shallow clone and work with normal clone

**Comment by harshithmullapudi** [comment]
Hey that's great. Do you mind helping me with more information (Right now it's hard to understand at which line this is breaking). 

1. Can you all some prints to that we can to understand that?
2. Also could you just try those lines outside in your terminal without the image so that we know for sure those lines are working?

@yash-ghelani-dsw

**Comment by skt7** [comment]
Hi @harshithmullapudi,

Let me try to rephrase and elaborate on the issue @yash-ghelani-dsw is pointing out. 

As we pointed out that the git shallow clone is causing the issue, we tried to replace it with simple clone command in the entrypoint.sh file of the "airbyte/normalization:0.2.6" image. We followed the following steps to achieve the same - 

1 - On the local copy of "airbyte/normalization:0.2.6" we changed the entrypoint.sh at two places - 
i) On line [67](https://github.com/airbytehq/airbyte/blob/8c300677408a27729e874d7fc76123aef43adbbb/airbyte-integrations/bases/base-normalization/entrypoint.sh#L67)
`git clone --depth ${GIT_HISTORY_DEPTH} --single-branch  "${GIT_REPO}" git_repo` 
to 
`git clone "${GIT_REPO}" git_repo`

ii) On line [71](https://github.com/airbytehq/airbyte/blob/8c300677408a27729e874d7fc76123aef43adbbb/airbyte-integrations/bases/base-normalization/entrypoint.sh#L71)
`git clone --depth ${GIT_HISTORY_DEPTH} -b "${GIT_BRANCH}" --single-branch  "${GIT_REPO}" git_repo` 
to 
`git clone  -b "${GIT_BRANCH}" --single-branch  "${GIT_REPO}" git_repo`

2 - We run the command `docker commit 4ffc6fb3c510 airbyte/normalization:0.2.6` to commit the changes. 

----

Post this we run the airbyte sync as usual then we tumbled upon the error that @yash-ghelani-dsw is pointing out - 

> 2022-08-22 06:59:41 normalization > bash: run: No such file or directory


Tracing back the same shows us that it fails while running - 

> 2022-08-22 06:59:41 INFO i.a.w.p.DockerProcessFactory(create):163 - Preparing command: docker run --rm --init -i -w /data/164/2/normalize --log-driver none --name normalization-normalize-164-2-hhisk --network host -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -e DEPLOYMENT_MODE=OSS -e AIRBYTE_ROLE= -e WORKER_ENVIRONMENT=DOCKER -e AIRBYTE_VERSION=0.39.36-alpha airbyte/normalization:0.2.6 run --integration-type postgres --config destination_config.json --catalog destination_catalog.json

We are not sure why we are getting this issue of file or directory not found whereas we haven't deleted any files but only modified the entrypoint.sh file, looking for your assistance on the same. Also if you can validate if we are moving in the right direction?

**Comment by harshithmullapudi** [comment]
Yeah your direction looks fine to me. Just trying to understand where that error is popping up. So if we can add some loggers to figure out which line is causing this that would be great.

**Comment by harshithmullapudi** [comment]
My guess is that git clone is causing that error `No file found` may be then we can dig deep there and try running the command local to see if that is working

**Comment by yash-ghelani-dsw** [comment]
> Yeah your direction looks fine to me. Just trying to understand where that error is popping up. So if we can add some loggers to figure out which line is causing this that would be great.

I have added logs below here:
[logs-164.txt](https://github.com/airbytehq/airbyte/files/9423011/logs-164.txt)

**Comment by yash-ghelani-dsw** [comment]
> My guess is that git clone is causing that error `No file found` may be then we can dig deep there and try running the command local to see if that is working

I have tried running following command on terminal:
`git clone <git_url> git_repo` 
And it works perfectly fine. It created a directory name git_repo and clone the git repository into same.

**Comment by harshithmullapudi** [comment]
Hey when you making changes in your normalisation image could you also add these `echo ""` lines to understand at which line this is breaking 

```bash
 if [[ -d git_repo ]]; then
      rm -rf git_repo
    fi
    # Make a shallow clone of the latest git repository in the workspace folder
    if [[ -z "${GIT_BRANCH}" ]]; then
      # Checkout a particular branch from the git repository
      echo "Running: git clone  --single-branch  \$GIT_REPO git_repo"
      git clone  --single-branch  "${GIT_REPO}" git_repo
    else
```

In the above code I don't think it even reached the `echo "Running: "` part so may be add echo before the if case which can help to debug

Otherwise we can get on a call to debug this you can reach out to me over slack

**Comment by yash-ghelani-dsw** [comment]
Hi @harshithmullapudi , We tried an alternative way to develop a local git repository. And it worked with airbyte. 
Instead of http server, we used git protocol provided by git itself. Using "git daemon" feature, we created a service out of it for linux.
 
Adding link for reference: [link](https://git-scm.com/book/en/v2/Git-on-the-Server-Git-Daemon) 

Url for Airbyte looks like: git://127.0.0.1/<repo_name>

Thank you for your assistance
